src/content_delegate.js

```diff
@@ -361,9 +361,20 @@ ContentDelegate.prototype.onDocumentViewSubmit = function (event) {
       // dcd (and presumably others) trigger this code path.
       var reader = new FileReader();
       reader.onload = function() {
+          let html=reader.result;
+          // check if we have an HTML page which redirects the user to the PDF
+          // this was first display by the Northern District of Georgia
+          // https://github.com/freelawproject/recap/issues/277
+          const redirectResult = Array.from(html.matchAll(/(?:window\.location\s=\s")(.*)(?:\")/g));
```

Comment:
The last time I had to do something similar, I used
`      if (match = /window\.location\s*=\s*["']([^"']+)["'];?/.exec(script.text))`
In particular:
. Why is the first non-capturing group present? It does not appear to serve any grouping function.
. Whitespace need not appear around the equals sign, so this is more robust with `\s*` rather than `\s`; and it could be more than one character long.
. JS strings need not be delimited by double-quotes, although my regexp may do the wrong thing if there is single-quoted string inside a double-quoted string, so I don't claim it's correct. (I don't think JS regexps have a great way to do this without two passes, unlike Python regexps with named groups  (`(?P=name)`). I guess you can do two passes if this were really important.)
. The final non-capturing group doesn't seem to serve a grouping function either. "When everything is a group, nothing is" — THE INCREDIBLES.
. What does the `/g` flag do on a search that is not a replacement?

Reply:
Adjusted along the lines of what's suggested.

On the /g flag, it finds all matches vs. stopping at a single match. Generally, /g is used with matchAll, but, in this particular case, we're not handling multiple results.

Reply:
Right. I guess I meant "Why is the /g appropriate here?" Seems like it's not?